aws-crt-sys/tests/lib.rs

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#[cfg(test)]
+mod tests {
+    use once_cell::sync::Lazy;
+    use std::sync::Mutex;
+    use std::ffi::CStr;
+    use std::env;
+
+    // CRT tests must be run serially because they affect global state
+    // As a result, we inject a mutex into each test using the CRT
+    // and while we're at it, we automate setup and teardown
+    static MUTEX: Lazy<Mutex<()>> = Lazy::new(Mutex::default);
+    macro_rules! with_crt {
+        ($test:block) => {
+            let _guard = match MUTEX.lock() {
+                Ok(guard) => guard,
+                Err(poisoned) => poisoned.into_inner(),
+            };
+            env::set_var("AWS_CRT_MEMORY_TRACING", "2");
+            unsafe {
+                aws_crt_init();
+                (|| $test)();
+                aws_crt_thread_join_all();
```

Comment:
See my other comment. You probably want to set a timeout before calling this, and report the error if it fails.
The way it is now, this will just lock your tests up forever if it fails